lib/moonshot/stack.rb

```diff
@@ -176,14 +176,35 @@ def get_stack(name)
       raise "Could not describe stack: #{name}"
     end
 
+    def upload_template_to_s3
+      raise 'The S3 bucket to store the template in is not configured.' unless @config.template_s3_bucket # rubocop:disable LineLength
+
+      s3_object_key = "#{@name}-#{File.basename(template.filename)}"
```

Comment:
For modern Moonshot users, this will always be "template", since the conventional location is [project root]/moonshot/template.yml. One thing I don't like about S3-based workflow is overwriting files loses you the context of historical stack state. Perhaps we should relate the filename to a timestamp or UUID, then we can see historical changes.

Reply:
See [here](https://github.com/acquia/moonshot/blob/master/lib/moonshot/stack.rb#L153).